### PR TITLE
Add safe transaction indexing section

### PR DESCRIPTION
**Pull request by moisses89**
Add blockchain indexing section

### CODE REVIEW

File: safe-core-api/safe-transaction-service.md [hunk 1]
```diff
@@ -23,6 +23,38 @@ Safe transaction service is a [Django](https://www.djangoproject.com/) app writt
...
+The indexer is running on `worker-indexer` by different periodic [tasks](https://github.com/safe-global/safe-transaction-service/blob/master/safe_transaction_service/history/tasks.py). 
+
+ERC20 and ERC721 are indexed using [eth_getLogs](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getlogs) filtered by the Transfer topic `keccak('Transfer(address,address,uint256)')`.
+
+Safe creation, executed transactions, configuration changes and onchain confirmations are indexed at different way depending if the chain is L1 or L2. 
+
+For L1 chains the indexer call tracing methods, for oldest blocks [trace_filter](https://openethereum.github.io/JSONRPC-trace-module#trace_filter) is used filtered by singleton address of safe contracts and for latest blocks [trace_block](https://openethereum.github.io/JSONRPC-trace-module#trace_block). The number of considered latest blocks is defined in `ETH_INTERNAL_TXS_NUMBER_TRACE_BLOCKS`. The environment variables used by indexing are defined [here](https://github.com/safe-global/safe-transaction-service/blob/master/config/settings/base.py#L433).
```
Comment (Uxio0) [inline safe-core-api/safe-transaction-service.md:34]:
```suggestion
For L1 chains the indexer calls tracing methods, for oldest blocks [trace_filter](https://openethereum.github.io/JSONRPC-trace-module#trace_filter) is used filtering by singleton address of Safe contracts and for latest blocks [trace_block](https://openethereum.github.io/JSONRPC-trace-module#trace_block) is used, as `trace_filter` takes longer to return updated information. `trace_block` will be used if the block depth is lower than `ETH_INTERNAL_TXS_NUMBER_TRACE_BLOCKS`. The environment variables used by indexing are defined [here](https://github.com/safe-global/safe-transaction-service/blob/master/config/settings/base.py#L433). 
```

File: safe-core-api/safe-transaction-service.md [hunk 2]
```diff
@@ -23,6 +23,38 @@ Safe transaction service is a [Django](https://www.djangoproject.com/) app writt
...
+
+Safe creation, executed transactions, configuration changes and onchain confirmations are indexed at different way depending if the chain is L1 or L2. 
+
+For L1 chains the indexer call tracing methods, for oldest blocks [trace_filter](https://openethereum.github.io/JSONRPC-trace-module#trace_filter) is used filtered by singleton address of safe contracts and for latest blocks [trace_block](https://openethereum.github.io/JSONRPC-trace-module#trace_block). The number of considered latest blocks is defined in `ETH_INTERNAL_TXS_NUMBER_TRACE_BLOCKS`. The environment variables used by indexing are defined [here](https://github.com/safe-global/safe-transaction-service/blob/master/config/settings/base.py#L433). 
+
+For L2 chains the indexing is by events with [eth_getLogs](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getlogs) method with the corresponding topics.  
+
+From safe creation the transaction service is storing each contract change on `SafeStatus` model as nonce, owners... and the last current status of a Safe in `SafeLastStatus`.
```
Comment (Uxio0) [inline safe-core-api/safe-transaction-service.md:38]:
```suggestion
From Safe creation the transaction service is storing each contract change on `SafeStatus` model as `nonce`, `owners`... and the last current status of a Safe in `SafeLastStatus`, for database easy access and optimization.
```

File: safe-core-api/safe-transaction-service.md [hunk 3]
```diff
@@ -23,6 +23,38 @@ Safe transaction service is a [Django](https://www.djangoproject.com/) app writt
...
+  "currentBlockNumber": 9773327, // Last block on blockchain
+  "erc20BlockNumber": 9773326,   // Last block indexed for erc20/721 events
+  "erc20Synced": true,
+  "masterCopiesBlockNumber": 9773327, // Last block indexed for executed transactions, ether transfers, configuration changes...
+  "masterCopiesSynced": true,
+  "synced": true
+}
+```
+
+**Reorgs handling**
```
Comment (Uxio0) [inline safe-core-api/safe-transaction-service.md:55]:
```suggestion
### Reorgs handling  
```

File: safe-core-api/safe-transaction-service.md [hunk 4]
```diff
@@ -23,6 +23,38 @@ Safe transaction service is a [Django](https://www.djangoproject.com/) app writt
...
+  "erc20BlockNumber": 9773326,   // Last block indexed for erc20/721 events
+  "erc20Synced": true,
+  "masterCopiesBlockNumber": 9773327, // Last block indexed for executed transactions, ether transfers, configuration changes...
+  "masterCopiesSynced": true,
+  "synced": true
+}
+```
+
+**Reorgs handling**  
+When indexed every block is marked as not confirmed unless it has some depth (configured via ETH_REORG_BLOCKS environment variable). Not confirmed blocks are checked periodically to check if the blockchain blockHash for that number changed before it reaches the desired number of confirmations, if that's the case, all blocks from that block and the transactions related are deleted and indexing is restarted to the last confirmed block.
```
Comment (Uxio0) [inline safe-core-api/safe-transaction-service.md:56]:
```suggestion
Every block is marked as `not confirmed` during indexing unless it has some depth (configured via `ETH_REORG_BLOCKS` environment variable). Not confirmed blocks are checked periodically to check if the blockchain `blockHash` for that number changed before it reaches the desired number of confirmations, if that's the case, all blocks from that block and the transactions related are deleted and indexing is restarted to the last confirmed block.

**Note:** No offchain signatures, transactions or messages are lost on this process. Only onchain data is removed.
```

File: safe-core-api/safe-transaction-service.md [hunk 5]
```diff
@@ -23,6 +23,38 @@ Safe transaction service is a [Django](https://www.djangoproject.com/) app writt
...
+
 ## Offchain Transaction Signatures
 The transaction service can collect offchain transaction signatures, allowing the owners to share their signatures to reach the required threshold before executing a transaction and spending less gas than onchain approvals.
```
Comment (Uxio0) [inline safe-core-api/safe-transaction-service.md:59]:
```suggestion
Safe Transaction Service can collect offchain transaction signatures, allowing the owners to share their signatures to reach the required threshold before executing a transaction and spending less gas than onchain approvals.
```

**Review by Uxio0 — CHANGES_REQUESTED** [review]

**Comment by Uxio0** [comment]
Closes https://github.com/safe-global/safe-transaction-service/issues/1655

File: safe-core-api/safe-transaction-service.md [hunk 6]
```diff
@@ -23,8 +23,42 @@ Safe transaction service is a [Django](https://www.djangoproject.com/) app writt
...
+  "erc20BlockNumber": 9773326,   // Last block indexed for erc20/721 events
+  "erc20Synced": true,
+  "masterCopiesBlockNumber": 9773327, // Last block indexed for executed transactions, ether transfers, configuration changes...
+  "masterCopiesSynced": true,
+  "synced": true
+}
+```
+
+### Reorgs handling  
+Every block is marked as `not confirmed` during indexing unless it has some depth (configured via `ETH_REORG_BLOCKS` environment variable). Not confirmed blocks are checked periodically to check if the blockchain `blockHash` for that number changed before it reaches the desired number of confirmations, if that's the case, all blocks from that block and the transactions related are deleted and indexing is restarted to the last confirmed block.
```
Comment (tanay1337) [inline safe-core-api/safe-transaction-service.md:56]:
```suggestion
Every block is marked as `not confirmed` during indexing unless it has some depth (configured via the `ETH_REORG_BLOCKS` environment variable). Unconfirmed blocks are checked periodically to see if the blockchain `blockHash` for that number changed before it reaches the desired number of confirmations. If that's the case, all blocks from that block and related transactions are deleted, and indexing is restarted to the last confirmed block.
```

File: safe-core-api/safe-transaction-service.md [hunk 7]
```diff
@@ -23,8 +23,42 @@ Safe transaction service is a [Django](https://www.djangoproject.com/) app writt
...
+
+**Note:** No offchain signatures, transactions or messages are lost on this process. Only onchain data is removed.
```
Comment (tanay1337) [inline safe-core-api/safe-transaction-service.md:58]:
```suggestion
**Note:** No offchain signatures, transactions, or messages are lost in this process. Only onchain data is removed.
```

**Review by tanay1337 — CHANGES_REQUESTED** [review]
Nicely done, @moisses89 🥂

**Review by tanay1337 — APPROVED** [review]

**Pull request merged**